pandas/core/generic.py

```diff
@@ -5134,12 +5135,15 @@ def tail(self: FrameOrSeries, n: int = 5) -> FrameOrSeries:
     @final
     def sample(
         self: FrameOrSeries,
-        n=None,
-        frac=None,
-        replace=False,
-        weights=None,
-        random_state=None,
-        axis=None,
+        n: Optional[int] = None,
+        frac: Optional[float] = None,
+        replace: bool_t = False,
+        weights: Optional[Union[str, ArrayLike]] = None,
+        random_state: Optional[
+            Union[int, ArrayLike, np.random.BitGenerator, np.random.RandomState]
```

Comment:
move this definition to _typing (call it RandomState i think is ok) or RandomStateType if it conflicts)